File: infrastructure/src/main/java/com/study/platform/domain/apply/infrastructure/ApplyMapper.java
package com.study.platform.domain.apply.infrastructure;

import com.study.platform.domain.apply.model.Apply;
import org.springframework.stereotype.Component;

@Component
public class ApplyMapper {

    public ApplyJpaEntity toEntity(Apply apply) {
        return ApplyJpaEntity.builder()
                .id(apply.getId())
                .post(apply.getPost())
                .applicant(apply.getApplicant())
                .message(apply.getMessage())
                .status(apply.getStatus())
                .build();
    }

    public Apply toDomain(ApplyJpaEntity entity) {
        return Apply.reconstitute(
                entity.getId(),
                entity.getPost(),
                entity.getApplicant(),
                entity.getMessage(),
                entity.getStatus(),
                entity.getCreatedAt(),
                entity.getUpdatedAt()
        );
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/apply/infrastructure/ApplyMapper.kt
package com.study.platform.domain.apply.infrastructure

import com.study.platform.domain.apply.model.Apply
import org.springframework.stereotype.Component

@Component
class ApplyMapper {

    fun toEntity(apply: Apply): ApplyJpaEntity {
        val entity = ApplyJpaEntity(
            id = apply.id,
            post = apply.post,
            applicant = apply.applicant,
            message = apply.message,
            status = apply.status
        )
        if (apply.createdAt != null) {
            entity.restoreTimestamps(apply.createdAt, apply.updatedAt)
        }
        return entity
    }

    fun toDomain(entity: ApplyJpaEntity): Apply = Apply.reconstitute(
        id = entity.id,
        post = entity.post,
        applicant = entity.applicant,
        message = entity.message,
        status = entity.status,
        createdAt = entity.createdAt!!,
        updatedAt = entity.updatedAt
    )
}
